### ISSUE TITLE
`--print_config` preserve original order

### ISSUE THREAD

**Issue by stasbel** [omni-us/jsonargparse#46]
Current yaml dump of config parser is in an unsorted order.
Sorted arguments order is much more readable for the user.

**Comment by mauvilsa** [comment]
Internally jsonargparse uses PyYAML and from what I understand by default it dumps sorted, see https://stackoverflow.com/a/55171433/2732151. Unless for some reason you are using some old version of PyYAML which does not sort it should be sorted. Can you include an example in which it is observed that it is not sorted? I don't know but the sort order could be affected by locale settings.

**Comment by stasbel** [comment]
sorry, wrong call
I mean the reverse thing: current yaml dump is in sorted order, which completely mess up the arguments order
imaging having three arguments related to something, like:
```python3
parser.add_argument('--a-model')
parser.add_argument('--b-model')
parser.add_argument('--c-model')
```
those 3 arguments would not appear together in `--print_config` dump

**Comment by mauvilsa** [comment]
I will look into this.

**Comment by mauvilsa** [comment]
In the current version in master the order is now preserved.

**Comment by mauvilsa** [comment]
I have just released v3.8.0 which includes the preservation of order.
